packages/lib/src/toolbar/controls/hooks.ts

```diff
@@ -0,0 +1,44 @@
+import type { ElementProps, FloatingContext } from '@floating-ui/react';
+import { useClickOutside, useKeyboardEvent } from '@react-hookz/web';
+import type { FocusEvent } from 'react';
+import { useCallback, useMemo } from 'react';
+
+/* Custom dismiss interaction hook for Floating UI widgets.
+ * Reduces bundle size (~16 kB) by replacing:
+ * - `useDismiss` (Escape key, click outside)
+ * - `FloatingFocusManager` (focus out) */
+export function useFloatingDismiss(context: FloatingContext): ElementProps {
+  const { refs, onOpenChange } = context;
```

Comment:
This is the kind of situation where I find that the `on...` convention for handlers does not hold well.

Oh well.

Reply:
Agreed, I could pass `toggle` directly, but it felt unnecessarily cumbersome.